godel_surface_detection/src/detection/surface_detection.cpp

```diff
@@ -268,8 +270,109 @@ void SurfaceDetection::get_region_colored_cloud(sensor_msgs::PointCloud2& cloud_
+  mesh_markers_.markers.clear();
+  meshes_.clear();
+
+  // Ensure cloud ptr is not empty
+  if (full_cloud_ptr_->empty())
+    return false;
+
+  // Create Processing Cloud
+  pcl::PointCloud<pcl::PointXYZ>::Ptr process_cloud_ptr (new pcl::PointCloud<pcl::PointXYZ>());
+  process_cloud_ptr->header = full_cloud_ptr_->header;
+
+  // Subsample full cloud into processing cloud
+  for(const auto& pt : full_cloud_ptr_->points)
+  {
+    int q = rand() % DOWNSAMPLE_NUMBER;
+    if (q ==0)
+    {
+      if(pt.x !=0.0 && pt.y!=0.0 && pt.z !=0.0 && pcl::isFinite(pt))
+      {
+        process_cloud_ptr->push_back(pt);
+      }
+    }
+  }
+
+  // Segment the part into surface clusters using a "region growing" scheme
+  surfaceSegmentation SS(process_cloud_ptr);
+  region_colored_cloud_ptr_ = CloudRGB::Ptr(new CloudRGB());
+  std::vector <pcl::PointIndices> clusters = SS.computeSegments(region_colored_cloud_ptr_);
+  pcl::PointIndices::Ptr inliers_ptr(new pcl::PointIndices());
+
+  // Extract points from clusters into their own point clouds
+  for (int i = 0; i < clusters.size(); i++)
+  {
+    Cloud::Ptr segment_cloud_ptr = Cloud::Ptr(new Cloud());
+    pcl::PointIndices& indices = clusters[i];
+    if (indices.indices.size() == 0)
+      continue;
+
+    if (indices.indices.size() >= params_.rg_min_cluster_size)
+    {
+      inliers_ptr->indices.insert(inliers_ptr->indices.end(), indices.indices.begin(), indices.indices.end());
+
+      pcl::copyPointCloud(*process_cloud_ptr, indices, *segment_cloud_ptr);
+      surface_clouds_.push_back(segment_cloud_ptr);
+    }
+  }
+
+  // Remove largest cluster if appropriate
+  if (params_.ignore_largest_cluster && surface_clouds_.size() > 1)
+  {
+    int largest_index = 0;
+    int largest_size = 0;
+    for (int i = 0; i < surface_clouds_.size(); i++)
+    {
+      if (surface_clouds_[i]->points.size() > largest_size)
+      {
+        largest_size = surface_clouds_[i]->points.size();
+        largest_index = i;
+      }
+    }
+    surface_clouds_.erase(surface_clouds_.begin() + largest_index);
+  }
+
+  // Compute mesh from point clouds
+  for (int i = 0; i < surface_clouds_.size(); i++)
+  {
+    pcl::PolygonMesh mesh;
+    visualization_msgs::Marker marker;
+
+    if (apply_concave_hull(*surface_clouds_[i], mesh))
+    {
+      // Create marker from mesh
+      mesh_to_marker(mesh, marker);
+
+      // saving other properties
+      marker.header.frame_id = mesh.header.frame_id = surface_clouds_[i]->header.frame_id;
+      marker.id = i;
+      marker.color.a = params_.marker_alpha;
+
+      // Push marker to mesh_markers_
+      ROS_INFO_STREAM("Adding a marker for mesh with " + std::to_string(marker.points.size()) + " points");
+      mesh_markers_.markers.push_back(marker);
+
+      // Push mesh to meshes_
+      meshes_.push_back(mesh);
+    }
+    else
+      ROS_INFO_STREAM("Apply concave hull failed");
+  }
+
+  return true;
+}
+
+
+bool SurfaceDetection::find_surfaces_old()
```

Comment:
Unless there is a good reason to keep this around in the active code, we should dispense with the original functions.